packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+      walletClient,
+      account,
+      checkSignature: false,
+      verifyingContract: collection,
+      publicClient: this.getPublicClient(),
+      uid: uid,
+      collection: signerData.collection,
+      premintConfig: signerData.premint,
+    });
+  }
+
+  /**
+   * Delete premint
+   *
+   * Deletes existing premint
+   * @param settings.collection collection address
+   * @param settings.uid UID
+   * @param settings.walletClient viem wallet client to use to sign
+   *
+   */
+  async deletePremint({
+    walletClient,
+    uid,
+    account,
+    collection,
+  }: {
+    walletClient: WalletClient;
+    uid: number;
+    account?: Account | Address;
+    collection: Address;
+  }) {
+    const signatureResponse = (await this.get(
+      `${ZORA_API_BASE}signature/${
+        this.network.zoraBackendChainName
+      }/${collection.toLowerCase()}/${uid}`,
+    )) as PremintResponse;
+
+    const signerData = {
+      ...signatureResponse,
+      premint: {
+        ...convertPremint(signatureResponse.premint),
+        deleted: true,
+      },
+    };
+
+    return await this.signAndSubmitPremint({
+      walletClient,
+      account,
+      checkSignature: false,
+      verifyingContract: collection,
+      publicClient: this.getPublicClient(),
+      uid: uid,
+      collection: signerData.collection,
+      premintConfig: signerData.premint,
+    });
+  }
+
+  private async signAndSubmitPremint({
+    walletClient,
+    publicClient,
+    verifyingContract,
+    premintConfig,
+    uid,
+    account,
+    checkSignature,
+    collection,
+  }: {
+    publicClient: PublicClient;
+    uid: number;
+    walletClient: WalletClient;
+    verifyingContract: Address;
+    checkSignature: boolean;
+    account?: Address | Account;
+    premintConfig: PremintConfig;
+    collection: PremintResponse["collection"];
+  }) {
+    if (!account) {
+      account = walletClient.account;
+    }
+    if (!account) {
+      throw new Error("No account provided");
+    }
+
+    const signature = await walletClient.signTypedData({
+      account,
+      ...preminterTypedDataDefinition({
+        verifyingContract,
+        premintConfig,
+        chainId: this.chain.id,
+      }),
+    });
+
+    if (checkSignature) {
+      const [isValidSignature] = await publicClient.readContract({
+        abi: zoraCreator1155PremintExecutorImplABI,
+        address: this.getExecutorAddress(),
+        functionName: "isValidSignature",
+        args: [collection, premintConfig, signature],
+      });
+      if (!isValidSignature) {
+        throw new Error("Invalid signature");
+      }
+    }
+
+    const apiData = {
+      collection,
+      premint: encodePremintForAPI(premintConfig),
+      chain_name: this.network.zoraBackendChainName,
+      signature: signature,
+    };
+
+    let premint = await this.retries(() =>
+      this.post(`${ZORA_API_BASE}signature`, apiData),
+    );
+
+    return {
+      zoraUrl: `https://${
```

Comment:
i'd lean against including this url here, because what if it changes?   Also this assumes users want the zora url when maybe they just want to mint and surface the mint on their own site.

I'd lean towards just having this api do the minimal that is needed, which is to create the premint and upload it, and if someone wants to get the zora mint url, they can call a helper method:

`zoraPremintMintUrl(address, uid)`

Reply:
per our discussion, we will keep this return value, but rename to `zoraCollectUrl` or have a `urls` property with multiple urls under it